### ISSUE TITLE
Make new workshops / morning challenges under the foundersandcoders organisation

### ISSUE THREAD

**Issue by jsms90** [foundersandcoders/master-reference#205]
An attempt to solve multiple problems:
1. Making contribution easier
  + Collaboration should be the default - everyone in the `foundersandcoders` org would be able to assign themselves to issues & submit PRs, without needing the original author to grant them access rights
  + If the original author is no longer maintaining the repo, and responding to issues or PRs, the material could still be iterated upon by others
2. Encouraging a sense of shared ownership over course materials
  + Maybe we would see more contributions to our resources if they weren't owned by one individual
  + Easier to critique the material during S-G-Cs if you don't feel like it's a personal attack on anyone's work
3. Making our material easy to find 
  + https://github.com/foundersandcoders/master-reference/issues/43 is not a great long-term solution
  + If any material gets swapped in / swapped out of the `master-reference`, alumni could still find the material that they had delivered to them when they were students (easier than trawlling through `master-reference` commit history)
4. Online presence for FAC (just a "nice to have" :wink:)
  + Having a growing body of material that is all under the FAC name means that people outside the organisation may even start to contribute
  + Recognition that these materials are being created because of this school

In a way, it's quite important to allow mentors to feel a sense of ownership over the material they create. It's good to develop a sense of pride about your work. Other benefits of having them under your own profile:
+ recognition of authorship (useful when talking to employers?)
+ re-usability in non-FAC contexts

But:
+ Rules/guidelines could be introduced, if necessary? e.g. the original author can be considered the "maintainer" and therefore needs to approve PRs?
+ Original authors can always pin these repos to their personal github profile, and the contribution graphs can be evidence of their input
+ FAC is open source by definition - people would be more than welcome to re-use material in other contexts. In fact, having these re-used can only help the profile of the school

Creating them under the `foundersandcoders` org by default makes more sense imo.

**Comment by nelsonic** [comment]
No idea why this isn't _already_ how it's done. thanks for "_fixing_" this @jsms90 👍

**Comment by jsms90** [comment]
@nelsonic This isn't something that can be fixed by me.

I think an org-wide conversation may need to be had, considering the fact that this was already requested https://github.com/foundersandcoders/master-reference/issues/43#issuecomment-281318614 and concerns raised.

Even if people agree, I'm unlikely to be the one communicating this to FAC10. I assume this would be done by @bradreeder during their curriculum planning days?

And if FAC10 are going to be able to iterate effectively on the material that was delivered to them, then each individual from FAC9 (and members of some FACs before 9) would need to be asked to move their material to `foundersandcoders`.

**Comment by jsms90** [comment]
But a decision does need to be made before FAC10's curriculum planning days

**Comment by iteles** [comment]
I've suggested this multiple times, you'll see the question peppered in comments over the last few years. Totally agree with this 👍

**Comment by jsms90** [comment]
@iteles @bradreeder @nelsonic @sofer What is the best way to action this?

Proposal at the curriculum committee? - I don't think this should be london-specific!
Proposal at the business meeting? - Is the wider organisation particularly interested?
Just do it?

**Comment by des-des** [comment]
@jsms90 I would be happy to mandate that new workshops / morning challenges that get built go into the master ref. We could also move forward by politely asking people if we they can move their challenges into master ref, then we can re-asses after the next iteration of the curriculum. At that point getting everything moved in might seem less daunting

**Comment by Jbarget** [comment]
If they are going into master-reference are they under repo's called `workshops` and `morning-challenges`?

We should definitely enforce some conformity to their structure and layout, see https://github.com/foundersandcoders/master-reference/issues/173

**Comment by des-des** [comment]
I think they should go in master reference, although I am not sure where to put solutions.

**Comment by sofer** [comment]
In the past we have sometimes put solutions into separate branches.

**Comment by des-des** [comment]
@sofer yeah, I think its an okay solution, but will be hard to manage and source control, especially if all in master ref

**Comment by bradreeder** [comment]
Thanks for raising @jsms90 . I think I'm in favour of moving workshops / morning challenges to the `foundersandcoders` organisation and maintaining an archive of resources with hyperlinks to them here.

If I were to action this, I'd suggest first getting in contact with FAC9 about moving all of the resources that were used for this curriculum, and finishing the archive in #43 . That's a solid baseline for the next curriculum-planning.

Agreed that a wider discussion might need to happen first as it concerns the ownership of materials that currently belong to the profiles of members. The cut-off for this business meeting is too late, so I'd suggest moving the discussion to `london-programme` for the next one.

**Comment by jsms90** [comment]
@Jbarget @des-des Morning challenges and workshops are usually repositories in their own right. So if you were to merge them into the master reference, they would have to become folders. But I _definitely_ don't think they should be amalgamated with this repo.

Otherwise every single issues with any of the workshops / morning challenges would live under this repo. This will spiral out of control really easily and make issues much less likely to get picked up and solved.

I suggest that we just move existing workshops / morning challenges into the `foundersandcoders` organisation, as they are? Then, the only things that need to be kept up-to-date in the master reference are the links to each one.

This would also make it much easier to retire a repo from the master reference, but not necessarily delete it altogether. Whereas, if everything were in the master reference, we should probably delete that workshop, so that the repo didn't get too bloated with unnecessary stuff.

**Comment by des-des** [comment]
Okay, from my perspective, more repos is harder to manage, if want to check if there are open pulls that need to be nugded, looking in one repo is easier, right now we have a folder called fragments, not sure if having workshops in there will bloat the repo, only make it easier for maintainers to see what is there.

However, I am happy to keep things separate for now. Then reopen this discussion and contemplate a merge when the conversation around how we can use master reference for many schools is revisited.

**Comment by Jbarget** [comment]
I'd vote for separate repos under the `foundersandcoders` org

**Comment by jsms90** [comment]
Keeping each workshop in a separate repo:
+ _Issue tracking_ is the most important reason for me. **It is the single best way to make iterative improvements successful.**  
I don't want to have to trawl through issues that are irrelevant to me planning my week #216

+ Lots of workshops ask students to clone the repo. It makes more sense to me, especially when that repo has it's own package.json, for it to be clone-able in its own right (not as part of `master-reference`)
+ As a student, it's much easier to compare what is in a self-contained repo, to the project that I'm about to create. For example, in node week, if the workshop has a `public` folder, a `src` folder and a `test` folder, maybe that's way that I should structure my own code. Of course you can still structure a sub-folder of the `master-reference` this way. It's just not as clean of a comparison.
+ Students will need to pull from master and update their local version of the `master-reference` when they make edits to it as a mentor to the next cohort.  
But what if the latest version of the `master-reference` has deleted a workshop that you did as a student? Now you don't have an reference to the one that you completed as a student any more. Even if it's just been moved to `fragments`, it's not as easy for me to find on my local machine any more. I look back at the materials I was delivered in FAC9 very frequently. This just introduces an irritating complexity.
+ Bloating the `master-reference` is just a bit annoying. If it gets big over the years, that's just more unnecessary files to introduce to your local machine when you clone. Ok, a very minor point...but still.
+ Authors can pin the workshops that they created to their own profiles for their own reference or for outside FAC use. You can't do that if everything is under the `master-reference`
+ It's harder to re-use workshops in a non-FAC context (something I think we should be able to do) if everything is _completely_ tied to the structure of our course.

**Comment by des-des** [comment]
@jsms90 good points, I am not in total agreement

**Comment by jsms90** [comment]
This time, to avoid a mad scramble with pull requests the day before each week begins, @bradreeder is aiming to have FAC10's improvements done by the end of week 15. Curriculum planning is now in 1.5 weeks. This issue has been given `priority 1` and from the conversation above, I assume that those of us who are maintaining the master reference are in agreement?

I raised [this issue in `london-programme`](https://github.com/foundersandcoders/london-programme/issues/217) to get a sense of whether the community would support the idea. I probably should have made a proposal already, but I was waiting for responses on the discussion issue first. It is slightly awkward for me whenever coordinating anything with my own cohort. But I think it's quite important to get this done before FAC10's curriculum planning. In which case, we need to move on this issue very quickly.

@bradreeder @des-des could you indicate with a thumbs up / response whether you support this?

### Week 1
+ [x] [Accessibility workshop](https://github.com/foundersandcoders/web-accessibility)
+ [ ] [Accessibility challenge](https://github.com/njsfield/accessibility-challenge)
+ [x] [Learn git basics](https://github.com/foundersandcoders/learn-git-basics) (**fork** :disappointed:)
+ [ ] [CSS gallery challenge](https://github.com/njsfield/css-gallery-challenge)

### Week 2
+ [ ] [Learn QUnit](https://github.com/skibinska/learn-qunit)
+ [ ] [Fizz buzz](https://github.com/skibinska/fizzbuzz)
+ [ ] [Romanizer](https://github.com/skibinska/romanizer)
+ [ ] [Waterfall function workshop](https://github.com/RhodesPeter/waterfall-function-workshop)
+ [ ] [Pure functions easy testing](https://github.com/Jwhiles/pure-functions-easy-testing)

### Week 3
+ [ ] [API workshop](https://github.com/lucymonie/api-workshop)
+ [ ] [XHR workshop](https://github.com/skibinska/xhr-workshop)
+ [ ] [Workshop APIs](https://github.com/emilyb7/workshop-APIs)
+ [ ] [Workshop client-side design](https://github.com/eliascodes/workshop-client-side-design)
+ [ ] [Parallel function morning challenge](https://github.com/foundersandcoders/master-reference/blob/master/coursebook/week-3/morning-challenge.md)

### Week 4
+ [ ] [Node intro workshop](https://github.com/bradreeder/Node-Intro-Workshop)
+ [ ] [ES6 morning challenge](https://github.com/stevehopkinson/es6-challenge)
+ [ ] [tape testing](https://github.com/matthewglover/tape-testing)
+ [ ] [modularising a simple node server](https://github.com/shouston3/modules-challenge)

### Week 5
+ [ ] [server with shot and tape](https://github.com/njsfield/tdd-node-server-with-shot-and-tape)
+ [ ] [error handling workshop](https://github.com/njsfield/error-handling-workshop)
+ [ ] [node shell workshop](https://github.com/msachi/Node-Shell-Workshop) (fork)
+ [ ] [request module workshop](https://github.com/RhodesPeter/request-module-workshop)

### Week 6
+ [ ] [postgres workshop](https://github.com/FAC9/postgres-workshop)
+ [ ] [pg walkthrough](https://github.com/shiryz/pg-walkthrough)
+ [ ] [pg workshop](https://github.com/shiryz/pg-workshop)
+ [ ] [db morning challenge](https://github.com/shiryz/db-morning-challenge)

### Week 7
+ [ ] [hapi intro](https://github.com/msachi/hapi-intro)
+ [ ] [handlebars intro workshop](https://github.com/marisid/handlebars-intro-workshop)
+ [ ] [handlebars morning challenge](https://github.com/Jbarget/handlebars-morning-challenge)
+ [ ] [injector](https://github.com/SavageWilliam/injector) (fork)
+ [x] [hapi auth cookie workshop](https://github.com/foundersandcoders/hapi-auth-cookie-ws)

### Week 8
+ [x] [OAuth workshop](https://github.com/foundersandcoders/oauth-workshop)
+ [x] [JWT workshop](https://github.com/foundersandcoders/jwt_workshop)

**Comment by jsms90** [comment]
I know that some people have questions about why students can't just fork their repo, so that they can keep it under their personal profile.

I am very aware that everyone on FAC9 has poured a little of their souls into this material, and I believe that everyone should receive recognition for the hard work that they have put in & retain a certain level of creative control (if they wish).

My suggestion for the future is:
> + Course coordinators are "lead maintainers" of the master reference, and are each expected to fulfill [these](https://opensource.guide/best-practices/) responsibilities
> + The person who wrote a workshop or morning challenge is the "author".
>   + An author is automatically considered the owner of their workshop.
>   + If an author doesn't wish to maintain the material, this responsibility defaults to the course coordinators (jointly) during the running of the cohorts.  
>   + On the 1st curriculum planning day, this responsibility can be passed to the one who will be delivering it, or anyone else in the community who has contributed to it.
> + Each curriculum-related stop go continue action point is turned into an individual issue on the master reference / workshop / morning challenge. Once validated by the maintainer, students (or alumni via github request) can then volunteer to complete the changes
> + Create a more comprehensive CONTRIBUTING guide
> + Create AUTHOR.md for each workshop - see [leadership & governance](https://opensource.guide/leadership-and-governance/)

(Taken from #248, using some guidance on maintaining open source projects: https://opensource.guide)

Many of us forked the repos of previous FACers when creating the `master-reference` this time round (myself included!!) But surely we don't want to end up with 50 versions of a good thing? We should be collaborating on the same resource, because many heads are better than one, and our students should be learning from the sum total of all FACers past :wink: 

These materials were ultimately created for FAC and its students, because we all believe in the vision of this organisation. All of the good work that our cohort has poured in will only continue if FAC10+ are able to effect change :star2:

**Comment by Jwhiles** [comment]
Why don't you just fork the workshops into Founders and Coders?

**Comment by Jwhiles** [comment]
It takes less clicks, + preserves the original copies and record of authorship without any extra work.

**Comment by lucymonie** [comment]
Buck up John old chap

**Comment by jsms90** [comment]
@Jwhiles 
In the world of github, forking implies that there is an expectation of it getting merged down the line. Hence, **you cannot raise issues on a forked version**. If you cannot raise issues, you cannot communicate effectively and therefore you cannot successfully iterate on material.

If we cannot always count on the original author to be active in FAC and update the resource in a timely manner, then its not possible for the resource to be iterated upon or contributed to. We created this material out of our responsibility to give back to Founders and Coders & contribute to its curriculum. So it belongs within the organisation, where it will be easiest for current mentors to continue contributing to it.

**Comment by Jwhiles** [comment]
I don't think forking implies intention to Merge

+ in ur settings :) 

![image](https://cloud.githubusercontent.com/assets/20307225/26274004/e3d92508-3d35-11e7-9b42-aaea1f0c5cf9.png)

**Comment by Jwhiles** [comment]
as an example here is a workshop I have forked into founders and coders, it has issues
https://github.com/foundersandcoders/pure-functions-easy-testing

**Comment by emilyb7** [comment]
I'm kind of in two minds about this. I spent a lot of time on some of my workshops. I'm sure, if they continue to be used, FAC10 and future FACsters could do great things with them and make them loads better. But then it's all out of my control, and given that I need to get a job and all that it would be nice to have the original repos around and intact and to be able to demonstrate _my_ work. 

I don't see that forking is all that bad, even if it's not what ~~god~~ github intended forking to be for.

**Comment by jsms90** [comment]
My apologies. Clearly I was wrong and mis-remembering what had caused the problems. But I promise you that forks have caused problems during the current cohort: https://github.com/Jwhiles/pure-functions-easy-testing/pull/13. I am only trying to make sure that we, as an orgnisation, learn from our mistakes & don't repeat things that we know don't work with each new set of students.

Can I ask whether you find the idea of transferring the repo into `foundersandcoders` offensive? Perhaps you feel like this is your personal material, that is being taken away from you? Is the problem that you don't want people to see a forked version in your profile and have people assume that it was created by someone else, after putting so much time in?

**Comment by Jwhiles** [comment]
My thinking is broadly in line with @emilyb7 - having public repos which are intended as teaching resources will look very good to a prospective employer. Having forked tutorials from the coding bootcamp you attended doesn't look like anything at all. We shouldn't expect recruiters to interrogate these repos and examine the authorship records, because they won't.

**Comment by lucymonie** [comment]
@Jwhiles and @emilyb7 if you are really keen to keep your repos and keep them on your own account, it is totally possible to fork back your own version and snip the leash [as outlined here](http://stackoverflow.com/questions/16052477/delete-fork-dependency-of-a-github-repository). 

My feeling is that we owe Founders & Coders in so many ways - and anything we can do to make the work of the people running the curriculum (*for nothing*) easier, we should do. 

Further, I think you'll look back in a year and find quite a bit wanting with the material you created right out of a coding bootcamp! It think it will be your contemporaneous work and side projects that will be of interest to employers.

**Comment by stevehopkinson** [comment]
Just a quick comment: if the main problem with forking was that it was merging into the original repo by default, I believe that can be updated fairly easily in the repo settings: https://help.github.com/articles/setting-the-default-branch/

**Comment by msachi** [comment]
It is also possible to duplicate a repository without forking it: https://help.github.com/articles/duplicating-a-repository/

**Comment by lucymonie** [comment]
@msachi - of course! And it's perfect because then you can keep all your issues and commit history.

Problem solved.

**Comment by des-des** [comment]
I can write a script to merge another git repository into this one.

Ie you could point it a git repo, and it would pull in the workshop + commit history and put the whole repo in a folder inside fragments/workshops in this repo?

I still have a strong preference for _everything_ to be in `master-reference`, as it means you do not have to go looking for stuff.

**Comment by jsms90** [comment]
@des-des i thought you agreed with me in the end?
https://github.com/foundersandcoders/master-reference/issues/205#issuecomment-290988656

**Comment by jsms90** [comment]
Yeah, @Jwhiles @emilyb7 that's totally fair! If I'm coming across too harshly, it is only because this made coordination difficult last time :sweat_smile: and if FAC10 & FACN1 are to make iterative improvements to the curriculum, time is of the essence.

It seems like there are two options, for material which is owned by an individual, rather than by FAC as a whole:

1. **You consider this to be open source material.** Your repo is still "live" i.e. you welcome updates & improvements, but you want to retain control over updates to its content, so that you can still consider it "yours".
2. **The original version of the repo is just an artefact that you want to be able to show to employers.** You don't actually want updates & improvements, because you want to retain the version that you created alone.  
Then I agree with @msachi - please duplicate the repo, using the link above, so ties can be cut and we avoid any confusion.

If FAC is truly about "open source", the message that we sent to students in future should probably be about making online material that is going to be open to iteration. So I wouldn't suggest duplication to future cohorts. I would suggest either option 1 (above) or create it in `foundersandcoders`.

But to be completely clear, I think either option is totally justified for FAC9! Creating almost the entire `master-reference` from scratch was a pretty big feat. The details of ownership were not remotely clear before we all created this material, so I'm not at all surprised that there are differences of opinion now.

**Comment by emilyb7** [comment]
@jsms90 of course I'm pro open source and welcome updates. Option 1 is basically my current situation. Does this mean I can stay being the owner and no one will mind and everything is fine?

**Comment by jsms90** [comment]
@emilyb7 Yeah sure :smile: It would actually be awesome for members of FAC to be maintainers of their own open source material :tada: If students decide that a pre-existing workshop is the right thing to include at that point in the curriculum, I see no difference between a Node Girls -owned workshop and an Emily Bertwhistle -owned workshop :+1: :smile: 

But thinking from the perspective of the students, I do think it's worth asking ourselves _why_ FAC9 largely decided to fork other people's existing repos. I don't think anyone raised issues on the originals, for the author to respond to? I can only imagine that this was about not having enough autonomy.
+ Will requests be responded to in time?
+ Will issues be responded to kindly? Maybe it sounds funny, but raising issues can be really intimidating if a maintainer isn't welcoming.
+ Will PRs be merged in time?
+ Will updates be accepted at all? Will students have to fight with authors, because the author was not present in the classroom to see what went wrong?
+ There's also the situation where nothing is _wrong_ with the existing workshop, and the original owner may use it in other contexts (e.g. makemehapi), but it would be more beneficial for FAC students if things were done in a different order, there were no ES6 syntax in it, or whatever.

I've been pushing hard for resources to be FAC-owned, to [encourage a sense of shared ownership over course materials](https://github.com/foundersandcoders/master-reference/issues/205#issue-215698123) because students are trying to design a cohesive week / curriculum, not just a particular workshop.

Given your original reasoning:
> But then it's all out of my control, and given that I need to get a job and all that it would be nice to have the original repos around and intact and to be able to demonstrate my work.

all I would ask is that you think about what scale of changes you would accept? And give students as much advanced warning as possible, if you aren't happy with anything about the direction they choose.

Does that sound fair?

**Comment by jsms90** [comment]
Again, sorry about my tone @emilyb7 @Jwhiles. Having a wide range of member-created and member-owned resources is not necessarily a bad thing at all. In fact, it could be a pretty beautiful thing to emerge from the FAC community - a wide variety of open source material and open source maintainers. I should have a little more faith :wink: 

@njsfield @skibinska @RhodesPeter @msachi @marisid @lucymonie @bradreeder @shouston3 @shiryz @tbtommyb @denesnori @SavageWilliam
I think it's only fair that you guys are presented with the same options, since I basically didn't present it like you guys had a choice before.

I'm going to raise a new issue, asking everybody to make an active choice in terms of whether they want to be an owner and/or maintainer. I really hope you guys can manage this before FAC10 get started on curriculum planning next week.

**Comment by jsms90** [comment]
Perhaps this has been closed prematurely? FAC10+ & FACN1+ may need to be pointed to this during their curriculum planning / mentorship
